src/amo/components/AddonRecommendations/index.js

```diff
@@ -29,53 +28,33 @@ import type { DispatchFunc } from 'core/types/redux';
 import './styles.scss';
 
 export const TAAR_IMPRESSION_CATEGORY = 'AMO Addon / Recommendations Shown';
-export const TAAR_COHORT_COOKIE_NAME = 'taar_cohort';
 export const TAAR_COHORT_DIMENSION = 'dimension4';
-export const TAAR_COHORT_INCLUDED: 'TAAR_COHORT_INCLUDED' =
-  'TAAR_COHORT_INCLUDED';
-export const TAAR_COHORT_EXCLUDED: 'TAAR_COHORT_EXCLUDED' =
-  'TAAR_COHORT_EXCLUDED';
+export const TAAR_COHORT_INCLUDED = 'TAAR_COHORT_INCLUDED';
 export const TAAR_EXPERIMENT_PARTICIPANT = 'TAAR-LITE-AB';
 export const TAAR_EXPERIMENT_PARTICIPANT_DIMENSION = 'dimension5';
 
-export type CohortName =
-  | typeof TAAR_COHORT_INCLUDED
-  | typeof TAAR_COHORT_EXCLUDED;
-
 type Props = {|
   addon: AddonType | null,
   className?: string,
-  cookie: typeof defaultCookie,
   dispatch: DispatchFunc,
   errorHandler: ErrorHandlerType,
   i18n: I18nType,
-  randomizer: () => number,
   recommendations: Recommendations | null,
   tracking: typeof defaultTracking,
 |};
 
 export class AddonRecommendationsBase extends React.Component<Props> {
   static defaultProps = {
-    cookie: defaultCookie,
-    randomizer: Math.random,
     recommendations: null,
     tracking: defaultTracking,
   };
 
   componentDidMount() {
-    const { addon, cookie, randomizer, recommendations, tracking } = this.props;
-
-    // Set a cohort for the experiment.
-    this.cohort = cookie.load(TAAR_COHORT_COOKIE_NAME);
```

Comment:
It looks like this turned out to be simpler than `react-ab` so that's nice. Simple is good. However, the next time we want to A/B test something we will have to start from scratch. I would like to see a more general purpose solution and it's fine to keep an issue open for that. I guess the tracker is https://github.com/mozilla/addons-frontend/issues/4262 ?

Any reason why the cohort/randomizer/GA code wasn't wrapped up in a higher order component or similar type of helper? At the very least, a function like `getCohort()` to wrap up the cookie logic would be useful.

Reply:
There was no particular reason that it wasn't encapsulated in an HOC, other than that I didn't think to do it. We actually started with an implementation that used `react-ab`, and then I realized that we actually didn't need that at all. Then it just kind of evolved from there. Agreed that in the future, if we need to do something similar, we should try to abstract it, and we can always look through the history of this component to see what we did before.

Reply:
I second Bob's comment. We decided to keep things simple and we did not know what to expect from this experiment. Now that it worked, it would be interesting to create a HOC for further use. We should probably keep this idea in an issue.